framework/include/utils/ChainedReal.h

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "libmesh/libmesh_common.h"
+#include "metaphysicl/metaphysicl_version.h"
+
+namespace MetaPhysicL
+{
+#if METAPHYSICL_MAJOR_VERSION < 1
```

Comment:
I appreciate the extra backwards compatibility, but can we not rely on MetaPhysicL being up to date?

Reply:
I copied this from `DualReal.h`. I am not sure I completely get what you mean here. If we were relying n MetaPhysicL being up-to-date, wouldn't we just do `#include "metalphyscl/dualnumber_forward.h"` without the if-else guard? The way it is right now seems to be "not" relying on MetaPhysicL being up-to-date.